src/libcore/num/mod.rs

```diff
@@ -70,15 +70,15 @@ assert_eq!(size_of::<Option<std::num::", stringify!($Ty), ">>(), size_of::<", st
                 #[stable(feature = "nonzero", since = "1.28.0")]
                 #[inline]
                 pub const unsafe fn new_unchecked(n: $Int) -> Self {
-                    $Ty(NonZero(n))
+                    $Ty(unsafe { NonZero(n) })
```

Comment:
Maybe I missed something: why is the inner `unsafe` necessary here within the body of an `unsafe fn` ? Does const-eval cause an effect analogous to https://github.com/rust-lang/rfcs/pull/2585 ? Because I am seeing warnings about an unnecessary `unsafe` when I do bootstrap builds right now.

Seems like it would warrant at least a comment in the code...

Reply:
That [was discussed](https://github.com/rust-lang/rust/issues/55607#issuecomment-435549534) on the tracking issue, and [recently discussed to not do anymore](https://github.com/rust-lang/rust/issues/55607#issuecomment-445882296). The revert PR is https://github.com/rust-lang/rust/pull/56706

Reply:
Ah okay thanks